fplll/nr/matrix.h

```diff
@@ -73,16 +76,18 @@ template <class T> class MatrixRow
     row.addmul_si_2exp(v.row, x, expo, n, tmp);
   }
 
-  /** Computes the dot product between two rows of a Matrix */
-  inline void dot_product(T &result, const MatrixRow<T> &v0, int n) const
+  /** Computes the truncated dot product between two rows of a Matrix.
+   * Constraint: n > beg.
+   */
+  inline void dot_product(T &result, const MatrixRow<T> &v0, int beg, int n) const
```

Comment:
Ah, so this PR modifies the interface of dot_product.

Reply:
Yes, and I realize this is not a good idea. There is obviously a way to keep a function `dot_product(T &, const MatrixRow<T> &, int n)` by specializing `dot_product(T &, const MatrixRow<T> &, 0, int n)`. This had the advantage to not break the API, especially for fpylll.

Reply:
I have reintroduced the previous way to call `dot_product`in commit 5330e00.